OpenSim/Simulation/SimbodyEngine/Test/testJoints.cpp

```diff
@@ -2226,6 +2231,42 @@ void testAutomaticJointReversal()
     ASSERT_EQUAL(accErr, 0.0, sqrt(integ_accuracy));
 }
 
+void testUserJointReversal()
+{
+    using namespace OpenSim;
+
+    cout << "\n==========================================================="
+         << "\n Test joint reversal set by user in old model file"
+         << "\n==========================================================="
+         << endl;
+
+    // Open model.
+    auto model = Model("double_pendulum_testReverse.osim");
+    //model.finalizeFromProperties();
+    model.initSystem();
+
+    // The topology is specified as follows:
+    //     [ground] <- (pin1) <- [rod1] <- (pin2) <- [rod2]
+    // but the "reverse" element of "pin2" is "true" so, after deserialization,
+    // the topology should be the following:
+    //     [ground] <- (pin1) <- [rod1] -> (pin2) -> [rod2]
```

Comment:
>  Is that what you intended?

Yes, but this test is ignorant of `MultibodyTreeMaker`. What you described is already tested in `testAutomaticJointReversal()`; the job of `testUserJointReversal()` is merely to ensure that the parent/child relationships are as expected after deserialization. If you want, I can add another model file where both "reverse" flags are "false", run a short simulation, and make sure that the coordinates from pin2 have opposite signs in the two models—but I believe that test would be redundant with the tests in `testAutomaticJointReversal()`.